src/main/java/de/tum/in/www1/artemis/web/rest/iris/IrisMessageResource.java

```diff
@@ -77,7 +83,12 @@ public ResponseEntity<List<IrisMessage>> getMessages(@PathVariable Long sessionI
     public ResponseEntity<IrisMessage> createMessage(@PathVariable Long sessionId, @RequestBody IrisMessage message) throws URISyntaxException {
         var session = irisSessionRepository.findByIdElseThrow(sessionId);
         irisSessionService.checkIsIrisActivated(session);
-        irisSessionService.checkHasAccessToIrisSession(session, null);
+        var user = userRepository.getUser();
+        irisSessionService.checkHasAccessToIrisSession(session, user);
+        var rateLimit = rateLimitService.getRateLimit(user);
+        if (rateLimit.isRateLimitExceeded()) {
+            throw new IrisRateLimitExceededException(rateLimit);
+        }
```

Comment:
Can you move this in some kind of `rateLimitService.checkRateLimitElseThrow(user);`?

Reply:
Good point. Done.